content/languages/ocaml/recipes.md

```diff
@@ -0,0 +1,36 @@
+---
+title: OCaml recipes for Codewars
+tags: [authoring, recipe, ocaml]
+---
+
+
+## Informative assertion messages
+
+Default assertion messages of OCaml OUnit are usually very bad. OUnit assertons accept two optional, named arguments:
+- `~printer`, used to stringify values presented by assertion messages.
+- `~msg`, used to provide additional information about failure, if necessary.
+
+With both arguments used, test output becomes more explicit:
+
+```ocaml
+"(square 3) should return 9" >:: (fun _ ->
+  let actual = square 3 in
+  assert_equal 9 actual ~printer: string_of_int ~msg: "Incorrect answer for n=3"
+)
+```
+
+```text
+Time: 1083ms Passed: 0Failed: 1Exit Code: 1
```

Comment:
Do we need this line? If you decide to keep it, then please insert spaces before `Failed` and `Exit Code`.

Reply:
Initially I wanted to add a screenshot because it would show quite well how bad the default messages are. But then it turned out that having screenshots in recipes would probably inflate amount of files significantly, especially when having two of them (dark mode and bright mode).  That's why I went with copy/paste of contents of the output panel.

Do you think it's OK this way? Or maybe unnecessary? Or maybe screenshots would be better?

I can remove the initial line, it's not exactly necessary.

Reply:
I prefer plain text for the output (easier to maintain and modify if necessary). The output looks better without the initial line.